modules/virt-creating-single-pvc-template-storage-pool.adoc

```diff
@@ -0,0 +1,78 @@
+
+:_content-type: PROCEDURE
+[id="virt-creating-single-pvc-template-storage-pool_{context}"]
+= Creating a storage pool using a `pvcTemplate` in a host path provisioner (HPP) custom resource.
+
+If you have a single large persistent volume (PV) on your node, you might want to virtually divide the volume and use one partition to store only the HPP volumes. By defining a `storagePool` using a `pvcTemplate` in the HPP custom resource, you can virtually split the PV into multiple smaller volumes, providing more flexibility in data allocation.
+
+A `pvcTemplate` matches the `spec` portion of a persistent volume claim (PVC).  For example:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: "iso-pvc"
+  labels:
+    app: containerized-data-importer
+  annotations:
+          cdi.kubevirt.io/storage.import.endpoint: "http://cdi-file-host.cdi:80/tinyCore.iso.tar"
+spec: <1>
+  volumeMode: Block
+  storageClassName: <any_storage_class>
+  accessModes:
+    - ReadWriteOnce
+       resources:
+         requests:
+           storage: 5Gi
+----
+<1> A `pvcTemplate` is the `spec` section of a PVC
+
+The operator creates a PVC from the PVC template for each node containing the HPP CSI driver. The PVC created from the PVC template consumes the single large PV, allowing the HPP to create smaller dynamic volumes.
+
+You can create any combination of storage pools. You can combine standard storage pools with storage pools that use PVC templates by using the  `storagePools` array.
+
+.Procedure
+
+. Create a YAML file for the CSI custom resource specifying a single `pvcTemplate`. For example:
++
+[source,terminal]
+----
+$ touch hostpathprovisioner_cr_pvc.yaml
+----
+
+. Edit the file. For example:
++
+[source,yaml]
+----
+apiVersion: hostpathprovisioner.kubevirt.io/v1beta1
+kind: HostPathProvisioner
+metadata:
+ name: hostpath-provisioner
+spec:
+ imagePullPolicy: IfNotPresent
+storagePools: <1>
+   - name: <any_name>
+     path: "</var/myvolumes>" <2>
+       pvcTemplate:
+       volumeMode: Block <3>
+       storageClassName: <any_storage_class> <4>
+       accessModes:
+       - ReadWriteOnce
+       resources:
+         requests:
+           storage: 5Gi <5>
+   workload:
+   nodeSelector:
+     kubernetes.io/os: linux
+----
+<1> The `storagePools` specification is an array to which you can add multiple entries.
+<2> Create directories under this node path. Read/write access is required. Ensure that the node-level directory (`/var/myvolumes`) is not on the same partition as the operating system. If it is, users of the volumes can potentially fill the operating system partition and cause the node to impact performance, become unstable, or become unusable.
+<3> `volumeMode` is optional and can be either `Block` or `Filesystem` but must match the provisioned volume format, if used. The default is `Filesystem`. If the `volumeMode` is `block`, the mounting pod creates an XFS filesystem on the block volume before mounting it.
```

Comment:
"XFS file system"

Two words "file system" for general references.

Reply:
Done.